x/evm/keeper/msg_server.go

```diff
@@ -269,7 +271,7 @@ func (k *Keeper) ApplyEvmMsg(ctx sdk.Context,
 	)
 
 	defer func() {
-		if commit && err == nil && resp != nil && !resp.Failed() {
```

Comment:
It's perfectly valid to have `err == nil` while the `vmError` is not empty. That will still result in a valid transaction.